packages/babel-types/src/builders/builder.ts

```diff
@@ -23,7 +22,9 @@ export default function builder<T extends t.Node>(
 
     let arg;
     if (i < countArgs) arg = args[i];
-    if (arg === undefined) arg = loClone(field.default);
+    if (arg === undefined) {
+      arg = Array.isArray(field.default) ? [] : field.default;
```

Comment:
in my search, all the default options are primitives except for a default of empty array

![types](https://user-images.githubusercontent.com/588473/112664145-0649b400-8e30-11eb-98e3-5a8bc81c27d3.png)

Reply:
We should throw if we see a non-empty array or an object, so that we don't accidentally break this assumption in the future.

Reply:
ah right, https://lodash.com/docs/#clone seems complicated, maybe we need to check what we do allow instead then given it could be anything?

so string/boolean/null/[]

Reply:
Yeah, we can just throw if `Array.isArray(def) ? def.length > 0 : def && typeof def === "object"`

Reply:
Added check in the other file! https://github.com/babel/babel/pull/13057/files#diff-36b14d40dba9fa52476c9907672a7a9e06a1da66dcc135100a918e956aadd04fR283-R289